alluxio/core.py

```diff
@@ -96,117 +98,28 @@ class AlluxioClient:
         },
 
     ]
-    >>> print(alluxio.read("s3://mybucket/mypath/dir/myfile"))
+    >>> print(alluxio_client.read("s3://mybucket/mypath/dir/myfile"))
     my_file_content
     """
 
     def __init__(
         self,
-        etcd_hosts=None,
-        worker_hosts=None,
-        options=None,
-        logger=None,
-        concurrency=64,
-        etcd_port=2379,
-        worker_http_port=ALLUXIO_WORKER_HTTP_SERVER_PORT_DEFAULT_VALUE,
-        etcd_refresh_workers_interval=120,
+        config: AlluxioClientConfig,
```

Comment:
I'm unsure whether it is a good practice to use a `config` class here. We would probably be careful as this is an interface change. I assume that the reason for this change is to avoid the long argument list for the `AlluxioClient` (although the users would still have to provide a long argument list for the `AlluxioClientConfig`).

Reply:
I checked a few other Python libraries:
1. scikit-learn. Take the random forest module (https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/ensemble/_forest.py#L1171) as an example, they just keep a long argument list.
2. presto-python-client. `PrestoRequest` (https://github.com/prestodb/presto-python-client/blob/master/prestodb/client.py#L131) also maintains a long argument list.
3. pymongo. `MongoClient` (https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/mongo_client.py#L141) keeps the argument list and uses `**kwargs` for other arguments.
4. Ray. `ray.tune` module (https://github.com/ray-project/ray/blob/master/python/ray/tune/tune.py) maintains a long list of arguments.

Reply:
To conclude,

1. It may not be an issue to maintain a long argument list. But for a long argument list, it might be a good practice to use `*` in the arguments (both ray and scikit-learn use that). https://stackoverflow.com/questions/57667742/what-does-represent-in-function-argument-list-in-python
2. After reviewing the configs, I feel some of them are client-related configs (e.g. etcd hosts, etcd ports, HTTP ports), and some are alluxio-specific configs (e.g. page size, hash nodes, etc.). Maybe it's better to differentiate those configs and construct these configs passed in as `**kwargs`. `pymongo` uses this approach (https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/mongo_client.py#L747, https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/client_options.py).

Reply:
From mongo client_options example, they are also putting all options into one options dict and then retrieve key values from it which is the main goal of this PR: putting all options in one place. I don't have strong preference about using dict or class, just feel there's might be two strength if we use a class instead of dict: it's more intuitive what's in the dict and what's not in the dict. And it's easy to convert to dict using: __dict__.

Reply:
And agree, we could do better by splitting different kind of options into their own scope

Reply:
The original reason is to have long list of argument in both alluxio-py and alluxiofs
We could discuss about whether to merge the two repos into one, otherwise too easy to break one of them